0.0.1/src/client.py

```diff
@@ -150,18 +321,62 @@ def _get_next_nonce(self, proxy_account_id: str) -> int:
             logger.error(f"Failed to get next nonce: {str(e)}", exc_info=True)
             raise
 
-    async def request_signature(self, proxy_account_id: str, request: SignatureRequest) -> str:
-        """Requests signature for given parameters"""
-        logger.debug(f"Requesting signature with: {request.dict()}")
+    async def _request_multi_action_signature(
+        self,
+        contract_id: str,
+        actions_json: str,
+        proxy_account_id: str,
+    ) -> Dict:
+        """
+        Request signature for multiple actions using MPC
+
+        Args:
+            contract_id: Target contract for the actions
+            actions_json: JSON string of actions to be signed
+            proxy_account_id: Account ID of the proxy contract
+
+        Returns:
+            Dict: Response from the proxy contract
+        """
         try:
-            result = await self._call_contract(proxy_account_id, request.dict())
-            success_value = result.status.get('SuccessValue')
-            logger.info(f"Successfully requested signature: {success_value}")
-            return self._decode_success_value(success_value)
+            # Get latest block hash if not provided
+            block_hash = await self._get_latest_block_hash()
+
+            # Get next nonce for the account
+            nonce = await self._get_next_nonce()
+
+            # Create signature request
+            signature_request = MultiActionSignatureRequest(
+                contract_id=contract_id,
+                actions_json=actions_json,
+                nonce=str(nonce),
+                block_hash=block_hash,
+                mpc_signer_pk=self.mpc_signer_pk,
+                account_pk_for_mpc=self.account_pk_for_mpc
+            )
+
+            self.env.add_system_log(
```

Comment:
Are there other cases for having this client class aware of the nearai bot env? Would be preferrable to avoid that coupling

Reply:
No, not needed. This can be replaced with a generic logger